labs/data_encoding_Services.py

```diff
@@ -0,0 +1,135 @@
+import pandas as pd
+from helpers.dslabs_functions import get_variable_types
+import numpy as np
+from math import pi, sin, cos
+import re
+
+def expand_loans(row, all_loans):
+    loans = row['Type_of_Loan'].split(',')
+    return [1 if loan in loans else 0 for loan in all_loans]
+
+def clean_loan_type(loan_type):
+    return loan_type.strip().replace('and ', '')
+
+def handle_loans(df):
+    all_loans = set(clean_loan_type(loan) 
+            for loan_list in df['Type_of_Loan'].dropna() 
+            for loan in loan_list.split(','))
+
+    expanded_rows = []
+    for loan_list in df['Type_of_Loan']:
+        loans = loan_list.split(',') if pd.notna(loan_list) else []
+        cleaned_loans = [clean_loan_type(loan) for loan in loans]
+        row = [1 if loan in cleaned_loans else 0 for loan in all_loans]
+        expanded_rows.append(row)
+    expanded_df = pd.DataFrame(expanded_rows, columns=list(all_loans))
+    df = pd.concat([df.reset_index(drop=True), expanded_df.reset_index(drop=True)], axis=1)
+    return df
+
+def transform_bools(df, keyword):
+    for col in df.columns:
+        if col.startswith(keyword):
+            df[col] = df[col].astype(int)
+    return df
+
+def clean_and_convert(s):
+    if pd.isna(s) or s == "nan":
+        return np.nan
+    cleaned = re.sub(r'[^0-9.]+', '', s)  # Remove non-numeric characters
```

Comment:
This also removes dashes, which are used to show negative numbers. I think the only negative number is -500, so that is fine as long as we handle all values above 122 years and 164 days [which is the oldest person registered](https://en.wikipedia.org/wiki/List_of_the_verified_oldest_people)

Reply:
Ok, I'll remove negative numbers first